projects/angular/src/forms/common/wrapped-control.ts

```diff
@@ -131,14 +128,25 @@ export class WrappedFormControl<W extends DynamicWrapper> implements OnInit, DoC
       this._id = this.controlIdService.id;
     }
 
-    if (this.ngControlService && this.ngControl) {
-      this.ngControlService.setControl(this.ngControl);
+    if (this.ngControlService && this._ngControl) {
+      if (!this.ngControlService.getControl()) {
+        this.ngControl = this._ngControl;
+        this.ngControlService.setControl(this.ngControl);
+        this.differ = this.differs.find(this._ngControl).create();
+      } else {
+        this.ngControl = this.ngControlService.getControl();
+        this.ngControlService.addAdditionalControl(this._ngControl);
```

Comment:
I am getting a bit confused how this works with multiple additional controls. It's probably only me, but I'd be happy to get some more details about it, thanks!

Reply:
Basically its multi level inheritance used here. WrappedControl is the base class for DateInputBase which is inturn base component for Single, Start & End datepicker components. DateRangeStartInput will be considered as Primary Control & DateRangeEndInput will be considered as additional Control (since primary control is already set i.e. DateRangeStart)

Reply:
I think I got it, thanks!